### PR TITLE
make: measure the op suite plus Granite in TEST_TYPE=perf

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
## What

`make tests TEST_TYPE=perf` invoked `spyre-perf-suite` with no `--mode`, so the
suite ran in its default `hf_adapters` mode and benchmarked HuggingFace models
only. The operator suite (matmul, pointwise ops, softmax, normalization, MLP,
and attention) was never measured by the CI perf run.

This changes the one invocation line in the `perf` target:

```
-	spyre-perf-suite --no-experimental --stacks torch-spyre \
+	spyre-perf-suite --mode fms --no-experimental --stacks torch-spyre \
 		--report "$(RESULTS_DIR)/report.txt"
```

The only added token is `--mode fms`. `--no-experimental` is kept (see below).

## Why this flag set

`--mode` and `--no-experimental` are two independent axes in
`run_benchmark.py`: `--mode` selects which suite runs, `--no-experimental`
gates the experimental ops inside it.

1. **`--mode fms`** routes the suite to its operator benchmarking path
   (`run_benchmarks`) instead of the models-only default (`run_hf_adapters`).
   This is the actual fix: the CI perf run now measures the operator suite.
2. **Granite stays measured.** Granite runs whenever `--no-granite` is absent,
   and we do not pass it. So the run covers the ops **and** Granite.
3. **`--no-experimental` is retained on purpose.** It skips only the
   experimental flash-attention op. That op is still unvalidated: a prior
   change that enabled it (#3718) was reverted (#3754) because the
   experimental flash-attention kernel was breaking the perf-suite CI run, and
   `spyre-perf-suite` has had no fix to that op since (the op definition and its
   gate are unchanged since the revert). Turning it on here would reintroduce
   the same breakage, so this PR does not.

So the perf run measures the operator suite **and** Granite, and deliberately
leaves the one known-broken experimental op out.

## Operator coverage (name reconciliation)

The operator suite's built-in op names do not line up one-to-one with the
generic op labels people ask for, so here is the mapping, verified in the
`spyre-perf-suite` source:

| Requested op | Covered by | Notes |
|---|---|---|
| `matmul` | `mul-QO-proj`, `mul-KV-proj`, `mul-MLP-proj` | all three projection aliases resolve to the `matmul` op |
| `pointwise_add` | `pointwise_add` | direct |
| `softmax` | `softmax` | direct |
| `rms_norm` | `layernorm` | the `layernorm` built-in calls `torch.nn.functional.rms_norm` |
| `layer_norm` | not present | genuinely absent from the built-in suite (distinct from `rms_norm` above) |
| `attention_gqa` | `attention_gqa` | direct; pinned to the `torch-spyre` stack |
| `sdpa` | `attention`, `attention_gqa` | both built-ins call `torch.nn.functional.scaled_dot_product_attention` |
| `mlp` | `mlp`, `mlp-linear-kn-t` | direct |
| `experimental.flash_attn` | not run | skipped by `--no-experimental`; still unvalidated, see "Why this flag set" |

`layer_norm` (true layer-norm, not RMS) is the only requested op the built-in
suite does not cover today. It is called out here rather than silently added:
authoring a new built-in op is a separate change.

## Scope

Single-line change to the `perf` target. No source files added, so the
Apache header and `import regex` conventions do not apply. `RESULTS_DIR` and
the `report.xml` assertion below the changed line are unchanged.

### CODE REVIEW

**Comment by ashokponkumar** [comment]
why do we want to run it on fms? it should be against torch-spyre / hf-adapters.

**Comment by Rafael-Sadykov** [comment]
We need to deploy the full list of spyre-perf-suite benchmarking, such as

- matmul
- pointwise_add
- softmax
- rms_norm
- layer_norm
- attention_gqa
- sdpa
- mlp
- experimental.flash_attn
- granite-3.3-8b-instruct_bs1_pl512_prefill_only_full_model
- granite-3.3-8b-instruct_bs1_pl512_prefill_only_one_block
- granite-3.3-8b-instruct_bs4_pl128_decode_only_full_model
- granite-3.3-8b-instruct_bs4_pl128_decode_only_one_block
- granite-3.3-8b-instruct_bs4_pl128_generate_full_model
- granite-3.3-8b-instruct_bs4_pl128_generate_one_block

We don't want to run in fms mode in the near future.

The backfill (e.g. #3691) command must consistently use:

```bash
spyre-perf-suite --no-experimental --stacks torch-spyre
```

This ensures historical runs exclude experimental benchmarks and use the same torch-spyre stack configuration, allowing results to be compared reliably across previous versions.

**Comment by HarikrishnanBalagopal** [comment]
CI is green on `47e31d7`: 160 checks pass, 0 fail. The 3 skipped checks are expected: `Perf benchmark` runs on real hardware through the internal perf pipeline (not in GHA, so it self-skips here), the pod-level-retry shard only fires on a prior failure, and the empty-suite report has nothing to report.

Ready for review. Summary of the change and the flag choice:

- The `perf` target invoked `spyre-perf-suite` with no `--mode`, so it ran the default `hf_adapters` mode and measured HuggingFace models only. The operator suite was never benchmarked by the CI perf run.
- The fix adds the single token `--mode fms`, which routes the run to the operator benchmarking path. Granite still runs (no `--no-granite`), so the run now covers the ops **and** Granite.
- `--no-experimental` is kept on purpose. It skips only the experimental flash-attention op, which is still unvalidated: a change that enabled it was reverted for breaking the perf-suite CI run, and there has been no fix to that op since. Turning it on here would reintroduce that breakage, so this PR leaves it off.

@Vikram123-byte @ashokponkumar as perf owners, could you take a look? Existing requested reviewers @anubhavjana @n-marion also welcome.

**Comment by HarikrishnanBalagopal** [comment]
Closing per maintainer guidance.

`--mode fms` was the wrong axis for this goal. It routes the run to the FMS
stack op suite, but the target is the torch-spyre stack. The per-op benchmarks
for the torch-spyre stack are meant to come from `--stacks torch-spyre`, not
from `--mode fms`, so this Makefile change does not measure what we want.

The Makefile line on `main` is already correct and should stay as is:

```
spyre-perf-suite --no-experimental --stacks torch-spyre --report "$(RESULTS_DIR)/report.txt"
```

The real fix belongs in `spyre-perf-suite`, not here: remove the `fms_` prefix
from the op names so `--stacks torch-spyre` surfaces the per-op list, and keep
the invocation consistent so the backfill for previous versions stays valid.

Target coverage the perf-suite change should yield:
`matmul, pointwise_add, softmax, rms_norm, layer_norm, attention_gqa, sdpa,
mlp, experimental.flash_attn`, plus the six granite-3.3-8b-instruct configs
(prefill_only / decode_only / generate, each full_model and one_block).

No torch-spyre change is needed. Closing.
